ShoppingM8-be/src/main/java/edu/pw/shoppingm8/product/api/ProductController.java

```diff
@@ -54,7 +59,9 @@
         @ApiResponse(code = 404, message = "If invalid id was provided")})
     @GetMapping("{id}")
     public ResponseEntity<ProductResponseDto> getProduct(@PathVariable Long listId, @PathVariable Long id) {
-        return ResponseEntity.ok(ProductResponseDto.of(productService.getProduct(listId, id)));
+        List list = listService.getList(listId);
+        listService.checkIfUserHasAccessTo(list);
```

Comment:
> 
> 
> General UI comments:
> 
>     * "See invitations" screen seems to be incomplete
> 
>     * Accepting invitation should refresh list view

"see invitations" meaning the tile at the end of the list, or the actual screen with invitations?

Reply:
The tile at the end of the list

Reply:
Tbh, my intention was to make it a little empty, like a list placeholder. Do you have any suggestions how to improve it?

Reply:
An empty tile could be ok, just different style (afaik there is dotted line and striped background, you could change it to solid). I don't remember text, but it could be something like "click to see invitations", something that would look like a clickable button

Reply:
I see background as solid... Maybe something with the opacity is not working on your device properly

Reply:
UI was improved. I also changed font size to be relative to screen size. I think that the stripes you've seen were caused by text being bigger than the screen. Please, try it now.

Reply:
My mistake, it was actually solid.
I think it would look better without the border and on white background. I leave the final decision up to you.

Reply:
I changed the tile slightly to make it almost transparent. Also border is much less visible, yet still existing. I think it looks okay now, better than before.